### PR TITLE
MAGN-9889 Preview Bubble should only appear when user intentionally hovers over a node

### PR DESCRIPTION
**Pull request by marimano**
### Purpose

- Preview bubble is shown if mouse stays on node longer than specified delay;
- Animations are removed;
- Preview bubble collapses when dragging is started;
- Code is enhanced;
- Tests passed.

### Task link

[MAGN-9889 Preview Bubble should only appear when user intentionally hovers over a node](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9889)

### Declarations

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any - *the PR changes behavior of UI rather than appearence, so there is no way to show it in snapshot*

### Reviewers

@mjkkirschner

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs [hunk 1]
```diff
@@ -59,6 +59,11 @@ public NodeViewModel ViewModel
             private set
             {
                 viewModel = value;
+                this.MouseLeave += (s, e) =>
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs]: hmmm... shouldn't we unsubscribe this somewhere else....

File: src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs [hunk 2]
```diff
@@ -10,21 +10,27 @@
...
 using Dynamo.Models;
+using System.Threading.Tasks;
 
 namespace Dynamo.UI.Controls
 {
-    // Event to be sent when PreviewControl goes into a stable 
-    // state (e.g. when all on-going storyboards have been completed).
+    /// <summary>
+    /// Event to be sent when PreviewControl goes into a stable 
+    /// state (e.g. when all on-going storyboards have been completed).
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs]: is this still true now that all the storyboard code has been removed?

File: src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs [hunk 3]
```diff
@@ -289,7 +292,7 @@ private void ResetContentViews()
         private void RunOnSchedulerSync(Action a, AsyncTaskCompletedHandler h)
         {
             var task = new DelegateBasedAsyncTask(scheduler, a);
-            task.ThenPost(h, DispatcherSynchronizationContext.Current);
+            task.ThenPost(h, SynchronizationContext.Current);
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs]: why this change?

File: src/DynamoSandbox/DynamoSandbox.csproj [hunk 4]
```diff
@@ -98,6 +98,7 @@
     <ProjectReference Include="..\DynamoUtilities\DynamoUtilities.csproj">
       <Project>{B5F435CB-0D8A-40B1-A4F7-5ECB3CE792A9}</Project>
       <Name>DynamoUtilities</Name>
+      <Private>False</Private>
```
Comment (mjkkirschner) [inline src/DynamoSandbox/DynamoSandbox.csproj]: what does this have to do with this PR? - and same below and for the other csproj at the top of these changes.

File: src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs [hunk 5]
```diff
@@ -101,7 +81,7 @@ public PreviewControl(NodeViewModel nodeViewModel)
             this.scheduler = nodeViewModel.DynamoViewModel.Model.Scheduler;
             this.nodeViewModel = nodeViewModel;
             InitializeComponent();
-            Loaded += OnPreviewControlLoaded;
+            Loaded += (s, e) => BeginNextTransition();
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs]: same concern - somewhere we can unsubscribe from this?

File: src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs [hunk 6]
```diff
@@ -499,143 +502,105 @@ private Size ContentToControlSize(Size size)
...
-        }
+            // show preview bubble only if mouse stays over node longer than delay time
+            // so that during quick mouse moving preview bubbles won't be shown
+            var delayTimer = new DispatcherTimer(DispatcherPriority.Normal)
+            {
+                Interval = TimeSpan.FromMilliseconds(500)
+            };
 
-        #endregion
+            nodeViewModel.OnMouseLeave += delayTimer.Stop;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs]: again for these subscriptions, do we need to unsubscribe from them at some point?

File: src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs [hunk 7]
```diff
@@ -499,143 +502,105 @@ private Size ContentToControlSize(Size size)
...
+            delayTimer.Tick += (obj, e) =>
+            {
+                Dispatcher.Invoke(ProcessFadeIn);
+                delayTimer.Stop();
+            };
 
-        #region Private Class Methods - Transition Helpers
+            await Task.Run(() => delayTimer.Start());
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs]: just curious since you are using tasks anyway, could we not just use task.delay?

**Comment by mjkkirschner** [comment]
@marimano I have a few questions, and I will continue looking at this a bit more as well.

File: src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs (hunk 3 above)
Comment (marimano) [inline src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs]: `Current` static member belongs `SynchronizationContext`, so resharper suggested this change

File: src/DynamoSandbox/DynamoSandbox.csproj (hunk 4 above)
Comment (marimano) [inline src/DynamoSandbox/DynamoSandbox.csproj]: when I try to debug and to see the state of an object, Visual studio says "DynamoUtilities is loaded twice" and so refuses to show state of any object

File: src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs (hunk 7 above)
Comment (marimano) [inline src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs]: as for me `Task.Delay` with cancellation mechanism would complicate it, that's why I didn't change Pavel's work here

**Comment by marimano** [comment]
hi @mjkkirschner, I updated this, please take a look again

**Comment by mjkkirschner** [comment]
@marimano LGTM
